docs/lsm/partial_compiles/partial_compiles.rst

```diff
@@ -0,0 +1,117 @@
+When building a new desired state, instead of replacing the entire desired state, we only replace specific ``ResourceSet``.
+Resources in ``ResourceSet`` can not depend on Resources in other ``ResourceSets``.
+
+To make this work, we have assign every Service Instance to a ``ResourceSet``, such that the set has no relations to any other ``ResourceSet``.
+
+In practice, we do this by putting all ``Resources`` in the ``ResourceSet`` of the parent entity.
+
+.. digraph:: resource_sets_generic_good
+    :caption: Resource Sets for Router example with 2 Routers with each 2 ports.
+
+    subgraph cluster_services {
+        "NullResource(name=r1)" [shape=rectangle];
+        "LifecycleTransfer(id=0)";
+        "LifecycleTransfer(id=0)" ->  "NullResource(name=r1)" 
+
+        "NullResource(name=r1-eth0)" [shape=rectangle];
+        "NullResource(name=r1-eth0)" -> "NullResource(name=r1)"
+        "LifecycleTransfer(id=2)" -> "NullResource(name=r1-eth0)"
+
+        label = "ResourceSet for Router r1(id=0)";
+        labelloc = "top";
+        color = "green";
+    }
+    subgraph cluster_services_2 {
+        "NullResource(name=r2)" [shape=rectangle];
+        "LifecycleTransfer(id=3)";
+        "LifecycleTransfer(id=3)" ->  "NullResource(name=r2)" 
+
+        "NullResource(name=r2-eth0)" [shape=rectangle];
+        "NullResource(name=r2-eth0)" -> "NullResource(name=r2)"
+        "LifecycleTransfer(id=4)" -> "NullResource(name=r2-eth0)"
+
+        label = "ResourceSet for Router r2(id=3)";
+        labelloc = "top";
+        color = "green";
+    }
+
+
+In addition to the ``ResourceSets`` used by individual services, there are also ``Resources`` that are not in any set.
+There ``Resources`` can be shared by multiple services, with the limitation that any compile that produces them, has to produce them exactly the same. 
+For more information see :ref:`Partial Compiles<partial_compile>`.
+
+Service Instance Selection
+++++++++++++++++++++++++++
+
+To have efficiency gains when recompiling, it is important to only build the model for all Service Instances that are in the ``ResourceSet`` we want to update and nothing else. 
+
+This selection is done automatically within ``lsm::all``, based on the relations set between the service bindings as explained above. 
+
+The underlying mechanism is that when we recompile for a state change on any Service Instance, we first search its owner by traversing :inmanta:attribute:`lsm::ServiceEntityBinding.relation_to_owner` until we reach a single owner. 
+Then we traverse back down the :inmanta:attribute:`lsm::ServiceEntityBinding.relation_to_owner` until we have all children.
+``lsm::all`` will only return these children and nothing else.
+
+Limitations
+-------------------
+
+1. When doing normal compiles, the model can very effectively find conflicts between services (e.g. using indexes), because it has an overview of all instances. 
```

Comment:
For this limitation, could we just refer to the [limitations section in the core docs](https://docs.inmanta.com/inmanta-service-orchestrator/6/model_developers/resource_sets.html#limitations)?

Reply:
You gave this a thumbs-up but I don't see a related change. Did you forget this or do you have a different opinion?